packages/wrangler/src/api/startDevWorker/ConfigController.ts

```diff
@@ -14,30 +13,43 @@ type Options = StartDevWorkerOptions;
 export class ConfigController extends Controller<ConfigControllerEventMap> {
 	config?: Options;
 
-	public set(input: Hook<Options, [Readonly<Options> | undefined]>) {
+	public set(input: Options) {
 		const config = unwrapHook(input, this.latest);
 
 		this.#updateConfig(config);
 	}
-	public patch(input: Hook<Partial<Options>, [Readonly<Options>]>) {
+	public patch(input: Partial<Options>) {
 		assert(
 			this.latest,
 			"Cannot call updateConfig without previously calling setConfig"
 		);
 
-		const partialConfig = unwrapHook(input, this.latest);
-
-		const config = deepmerge(this.latest, partialConfig, {
-			arrayMerge: (target, _source, _options) => target, // arrays are overridden, not concatenated (deepmerge default)
-		});
+		const config: Options = {
+			...this.latest,
+			...input,
+		};
 
 		this.#updateConfig(config);
 	}
 
 	latest?: Options;
 	#updateConfig(input: Options) {
+		const directory = input.directory;
+
+		this.config = {
+			directory,
+			build: {
+				moduleRules: [],
+				additionalModules: [],
+				define: {},
+				format: "modules",
+				moduleRoot: directory,
+				...input.build,
+			},
+			...input,
+		};
 		this.latest = input;
-		this.emitConfigUpdateEvent(this.latest);
+		this.emitConfigUpdateEvent(this.config);
```

Comment:
This looks like a bug fix? If so what is it fixing? Is there a relevant test that would fail if this got reverted?
What is the difference between `this.latest` and `this.config`?

Reply:
It is a bug fix of the startWorker programmatic interface only (not wrangler_dev or unstable_dev). There are tests in this PR that use the `worker.config` (which is a getter for this ConfigController.config) value and will fail if it is not set